docs/docs/integrations/scoring-reranking-models/3-vertex-ai-ranking.md

```diff
@@ -0,0 +1,90 @@
+---
+sidebar_position: 3
+---
+
+# Google Cloud Vertex AI Ranking API
+
+- [Google Cloud Vertex AI Ranking documentation](https://cloud.google.com/generative-ai-app-builder/docs/ranking)
+- [Google Cloud Vertex AI Ranking API description](https://cloud.google.com/generative-ai-app-builder/docs/reference/rest/v1/projects.locations.rankingConfigs/rank)
+
+
+### Introduction
+
+The Google Cloud Vertex AI Ranking API is a powerful tool that enhances search results by refining the relevance of
+retrieved documents to a given query. Unlike traditional search methods, it leverages advanced machine learning 
+algorithms to understand the semantic context of both the query and the documents, delivering more precise and relevant 
+results. By analyzing the semantic relationship between the query and each document, the API can reorder the candidate 
+documents based on their calculated relevance scores, ensuring that the most relevant results appear at the top of the 
+search results page.
+
+### Maven Dependency
+
+```xml
+<dependency>
+    <groupId>dev.langchain4j</groupId>
+    <artifactId>langchain4j-vertex-ai</artifactId>
+    <version>0.34.0</version>
+</dependency>
+```
+
+### Usage
+
+You can score a single string or `TextSegment` against a query 
+thanks to the `score(text, query)` and `score(segment, query)` methods.
+
+It is also possible to score several strings or `TextSegment`s against the query, 
+with the `scoreAll(segments, query)` method:
+
+```java
+VertexAiRankerScoringModel scoringModel = VertexAiRankerScoringModel.builder()
+    .projectId(System.getenv("GCP_PROJECT_ID"))
+    .projectNum(System.getenv("GCP_PROJECT_NUM"))
```

Comment:
What is project num?

Reply:
This API requires also the project _number_ (in addition to the project _id_).

You'll have to add it to your CI environment variables.

The project number can be found in the welcome dashboard in the cloud web console. or you can also run `$ gcloud projects describe langchain4`, and you'll find it in the output.

Reply:
It would probably make sense to add this info to the docu?